Functions/xASL_fsl_SetFSLdir.m

```diff
@@ -199,12 +204,16 @@
 
 
 %% Manage RootFSLDir
-if ~exist('RootWSLdir','var') || isnan(RootWSLdir)
+if ~exist('RootWSLdir','var')
     RootWSLdir = FSLdir; % default
 else
     RootWSLdir = fullfile(RootWSLdir, FSLdir); % for e.g. WSL
 end
 
+RootWSLdir = strtrim(regexprep(RootWSLdir,char(0),''));
```

Comment:
The issue I had here was a null character at the start of RootFSLdir (not a space - an actual NULL!). This is on a Mac with FSLDIR=/usr/local/fsl - no spaces or oddities in the path. I struggled to follow the code defining this so added this check to remove it from any of these paths. I hadn't seen xASL_adm_UnixPath - this would definitely be a better place to do this, although it would also be good to understand where the null character comes from in the first place.

Reply:
Yes, good point, if you can give me an example I can try to find the origin of this NULL (I use macOS as well).